modules/core/04-channel/keeper/export_test.go

```diff
@@ -7,9 +7,25 @@ package keeper
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
 	"github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
 )
 
+// StartFlushUpgradeHandshake is a wrapper around startFlushUpgradeHandshake to allow the function to be directly called in tests.
+func (k Keeper) StartFlushUpgradeHandshake(
+	ctx sdk.Context,
+	portID,
+	channelID string,
+	proposedUpgradeFields types.UpgradeFields,
+	counterpartyChannel types.Channel,
+	counterpartyUpgrade types.Upgrade,
+	proofCounterpartyChannel,
+	proofCounterpartyUpgrade []byte,
+	proofHeight clienttypes.Height,
+) error {
+	return k.startFlushUpgradeHandshake(ctx, portID, channelID, proposedUpgradeFields, counterpartyChannel, counterpartyUpgrade, proofCounterpartyChannel, proofCounterpartyUpgrade, proofHeight)
+}
```

Comment:
I did a quick check on this since it wasn't clear if having the function in a `_test.go` file would prevent it being used. 

I added the method sig to the expected keeper interface of `ChannelKeeper` in transfer and there was no compiler issues and you could call the method with no issues. So to me, it seems like just making it exported from the get go is the same thing

Reply:
wait, the function should not be added to the binary in non-test mode, this is in `go help build`:

> When compiling packages, build ignores files that end in '_test.go'.

the compiler as in the LSP that does checking or does a `go build` succeed? (the second case would sound like a serious bug :smile:)

Reply:
You sir, are correct. It was the LSP that has me misled. A `make install` fails rightly as you say!

Reply:
`startFlushUpgradeHandshake` should only be used during the upgrade handshake and thus does not need to be referenced outside of the `04-channel/keeper` package. Given that our versioning is dependent upon API breaking changes, it gives us more flexibility to handle fixes without incrementing our major version if this function is unexported. In my opinion, only the `Msg`'s and grpc handlers should be exported. It is unsafe to reference this function outside of the context it is called in (the upgrade message handlers). I think in general we should aim to limit our public API as it gives us maximum development flexibility while also establishing a clear public API external developers can rely on. Unless there's a reason to make the function public, I think it is healthy to default to leaving the function private